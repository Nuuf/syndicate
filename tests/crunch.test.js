/* eslint-disable @typescript-eslint/no-var-requires */
const syndicate = require('../dist');

test('crunches a syndicate root', () => {
  const root = syndicate.createRoot();
  syndicate.add(root, syndicate.create({}, 'A'));
  syndicate.add(root, syndicate.create({}, 'B'));
  syndicate.add(root, syndicate.create({}, 'C'));
  syndicate.add(root, syndicate.create({}, 'D'));
  syndicate.add(root, syndicate.create({}, 'E'));
  syndicate.add(root, syndicate.create({}, 'F'));
  syndicate.remove(root, syndicate.find(root, 'F'));
  syndicate.remove(root, syndicate.find(root, 'A'));
  syndicate.remove(root, syndicate.find(root, 'C'));
  syndicate.crunch(root);
  expect(JSON.stringify(root, null, 2)).toMatchSnapshot();
});
