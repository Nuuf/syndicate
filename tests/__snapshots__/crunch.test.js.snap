// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`crunches a syndicate root 1`] = `
"[
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":-1,\\\\\\"parentId\\\\\\":null,\\\\\\"index\\\\\\":0,\\\\\\"id\\\\\\":\\\\\\"root\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"root\\\\\\"},\\\\\\"childrenIndices\\\\\\":[1,2,3],\\\\\\"childrenIds\\\\\\":[\\\\\\"B\\\\\\",\\\\\\"D\\\\\\",\\\\\\"E\\\\\\"]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":1,\\\\\\"id\\\\\\":\\\\\\"B\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":2,\\\\\\"id\\\\\\":\\\\\\"D\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":3,\\\\\\"id\\\\\\":\\\\\\"E\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\"
]"
`;
